tracks/csharp/exercises/circular-buffer/mentoring.md

```diff
@@ -0,0 +1,55 @@
+### Reasonable solution
+
+#### using `Queue`
+This exercise aims to teach two concepts: queue, as an abstract data structure, and how to use the associated implementation `Queue<T>`.
+
+```csharp
+using System;
+using System.Collections.Generic;
+
+public class CircularBuffer<T>
+{
+    private readonly int _capacity;
+    private readonly Queue<T> _buffer;
+
+    public CircularBuffer(int capacity)
+    {
+        _capacity = capacity;
+        _buffer = new Queue<T>(capacity);
+    }
+
+    public T Read()
+    {
+        if (_capacity < 1) 
+            throw new InvalidOperationException("Circular buffer capacity needs to be greater than 0 !");
+        else 
```

Comment:
You could consider losing the `else` statement.

Reply:
yes, removed.